[2.0] Remove simple references leftovers

malarzm opened the pull request:
We need to decide whether we still want to call references using `storeAs="id"` "simple" or not (I would be in favour of doing so, feels more natural than "references stored as ids" however I may just be used to "simple references" term)

alcaeus commented:
I've started calling them "ID references", opposed to "DBRef references". Since there's no more mention to "simple" in the code, I'd like to avoid that term in the future since it might make it more complex to understand for newbies.

coudenysj commented:
I like the @alcaeus approach :)

malarzm commented:
Changed "simple references" occurrences to "ID references" as suggested :+!:

alcaeus commented on tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH499Test.php:
```diff
@@ -1,68 +1,68 @@
+        $this->dm->flush();
+        $this->dm->clear();
+
+        $collection = $this->dm->getDocumentCollection(__NAMESPACE__ . '\GH499Document');
```
You can use the `GH499Document::class` here.

malarzm replied:
I'm more wondering why is this marked as an addition, it should only be mapping changed...

alcaeus replied:
Looking at the [diff without whitespaces](https://github.com/doctrine/mongodb-odm/pull/1480/files?w=0) hides the lines, so I assume there were changes to the line endings.

malarzm replied:
Funny, the only thing that happened to this files was a rebase, will revisit my git settings and fix it :+1:

alcaeus commented on tests/Doctrine/ODM/MongoDB/Tests/Functional/Ticket/GH499Test.php:
```diff
@@ -1,68 +1,68 @@
+
+        $collection = $this->dm->getDocumentCollection(__NAMESPACE__ . '\GH499Document');
+
+        $a = $collection->findOne(array('_id' => new \MongoId($a->getId())));
```
Short array syntax

alcaeus commented on tests/Doctrine/ODM/MongoDB/Tests/Mapping/Driver/fixtures/yaml/TestDocuments.User.dcm.yml:
```diff
@@ -1,44 +1,44 @@
-  alsoLoadMethods:
-    doStuffOnAlsoLoad: unmappedField
-
+TestDocuments\User:
```
I see Windows line endings here.

malarzm commented:
@alcaeus I've fixed CRLF mess that I introduced

> I think the changes to the mapping in tests (changing simple to storeAs) could be backported to the master branch, what do you think?

I've split commits into fixing test mappings and removing leftovers for cherry picking 👍
